src/roles/common/tasks/fix-image-paths.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name: Get unzipped files dir
+  set_fact:
+    nuage_unzipped_files_dir: "{{ common.nuage_unzipped_files_dir }}"
+
+- name: Replace non-relative path
+  set_fact:
+    nuage_unzipped_files_dir: "{{ nuage_unzipped_files_dir | regex_replace('.+?(?=metroae_data)', '/') }}"
+  when: common.nuage_unzipped_files_dir | regex_search("/metroae_data/")
+
+- name: Replace relative path
+  set_fact: 
+    nuage_unzipped_files_dir: "{{ '/metroae_data/' + nuage_unzipped_files_dir }}"
+  when: common.nuage_unzipped_files_dir | regex_search("^[^/]")
+
+- name: Get stats of resulting path
+  stat: 
+    path: "{{ nuage_unzipped_files_dir }}"
+  register: sym
+
+- name: Check resulting path validity
+  assert:
+    that:
+    - sym.stat.exists
+    fail_msg: "MetroAE unzipped files directory path: {{ nuage_unzipped_files_dir }} is invalid"
+  ignore_errors: True

```

Comment:
Why ignore_errors: true?

Reply:
I thought it would be better when checking if the path is valid that only an error message would be printed out to be seen, as opposed to printing the error and halting all subsequent tasks. When discussing with Sanket he had placed the "ignore_errors: True" line at the end of the previous task in that file (get stats of resulting path), but I thought it would make more sense to place it the check resulting path validity task. This is because I don't think the stat module would error out if the directory path doesn't exist (in the get stats of resulting path task), but the check resulting path validity task definitely would.

If it's better to leave out the "ignore_errors: True" line, that should be a quick edit. Can you explain when it may be appropriate to have a "ignore_errors: True" in a task if that's the case?

Reply:
Assert should not have ignore_errors true. If any task before assert errors out, it can have ignore errors as assert will print out a nicer error message.